src/lifecycles/load.js

```diff
@@ -64,7 +66,7 @@ export function toLoadPromise(app) {
     })
     .catch(err => {
       handleAppError(err, app);
-      app.status = SKIP_BECAUSE_BROKEN;
+      app.status = err instanceof InternalError ? SKIP_BECAUSE_BROKEN : LOAD_ERROR;
```

Comment:
Only use the new error state _if_ the promise itself fails. Not if the load function doesn't return a promise. Also don't use the new error state if the promise itself isn't resolving with the proper stuff. In theory this should mean we only ever retry if the load promise itself fails.